src/intersections/rays.jl

```diff
@@ -11,49 +11,54 @@ The intersection type can be one of six types:
 5. overlap with colliding vectors (OverlappingOpposingRays -> Segment)
 6. do not overlap nor intersect (NoIntersection)
 =#
-function intersection(f, r1::Ray{N,T}, r2::Ray{N,T}) where {N,T}
-  a, b = r1(0), r1(1)
-  c, d = r2(0), r2(1)
+function intersection(f, ray1::Ray{N,T}, ray2::Ray{N,T}) where {N,T}
+  a, b = ray1(0), ray1(1)
+  c, d = ray2(0), ray2(1)
 
-  λ₁, λ₂, r, rₐ = intersectparameters(a, b, c, d)
+  # normalize points to gain parameters λ₁, λ₂ corresponding to arc lengths
+  len1, len2 = norm(b - a), norm(d - c)
+  b₀ = a + 1/len1 * (b - a)
+  d₀ = c + 1/len2 * (d - c)
+
+  λ₁, λ₂, r, rₐ = intersectparameters(a, b₀, c, d₀)
```

Comment:
Now in this case here you add b0 and d0...

Reply:
both intersecting objects have "critical" points where the length scale matters (`mayberound` is used)